Symfony/CS/Tokenizer/Tokens.php

```diff
@@ -879,7 +879,7 @@ public function isShortArray($index)
 
         $prevToken = $this[$this->getPrevMeaningfulToken($index)];
 
-        if ($prevToken->equalsAny(array(array(T_DOUBLE_ARROW), '=', '+', '(', '['))) {
+        if ($prevToken->equalsAny(array(array(T_RETURN), array(T_DOUBLE_ARROW), '=', '+', '(', '['))) {
```

Comment:
This is incomplete - probably as a result of testing for short arrays the wrong way.

It still fails for:
```php
<?php
function a()
{
    yield [
        'é' => 5,
        5 => 'é'
    ];
}
```